numpy/core/src/multiarray/dlpack.c

```diff
@@ -197,12 +196,18 @@ array_dlpack(PyArrayObject *self,
         return NULL;
     }
 
-    DLManagedTensor *managed = PyMem_Malloc(sizeof(DLManagedTensor));
-    if (managed == NULL) {
+    // ensure alignment
+    int offset = sizeof(DLManagedTensor) % sizeof(void *);
+    void *ptr = PyMem_Malloc(sizeof(DLManagedTensor) + offset +
+        (sizeof(int64_t) * ndim * 2) +
+        (sizeof(int64_t) * ndim * 2) % sizeof(void *));
```

Comment:
```suggestion
        (sizeof(int64_t) * ndim * 2));
```
Or is there a reason for padding?  It seems necessary only if we would want to make an array of these, but we don't.

Reply:
I guess introducing a `struct {DLManagedTensor tensor, uint64_t shape_strides[]}` would be clunky?  That would allow us to just use `sizeof()` and no modulo and no need to use `ptr + offset`.
(I.e. if you think it is nicer, could be cool, if not, don't worry.)

Reply:
> I guess introducing a `struct {DLManagedTensor tensor, uint64_t shape_strides[]}` would be clunky?

Yeah, I feel like it's a bit clunky.